packages/utils/src/conversions.ts

```diff
@@ -41,111 +42,128 @@ export function bigIntToHexadecimal(n: bigint): string {
 }
 
 /**
- * Converts a hexadecimal string to a bigint, ensuring the hex string starts with '0x'.
+ * Converts a hexadecimal string to a bigint. The input is interpreted as hexadecimal
+ * with or without a '0x' prefix. It uses big-endian byte order.
  * @param hex The hexadecimal string to convert.
  * @returns The bigint representation of the hexadecimal string.
  */
-export function beHexadecimalToBigint(hex: string): bigint {
+export function beHexadecimalToBigInt(hex: string): bigint {
     // Ensure the hex string starts with '0x'.
     const formattedHexString = hex.startsWith("0x") ? hex : `0x${hex}`
 
     return BigInt(formattedHexString)
 }
 
 /**
- * Converts a hexadecimal string to a bigint. Alias for beHexadecimalToBigint.
+ * Converts a hexadecimal string to a bigint. Alias for beHexadecimalToBigInt.
  * @param hex The hexadecimal string to convert.
  * @returns The bigint representation of the hexadecimal string.
  */
-export function hexadecimalToBigint(hex: string): bigint {
-    return beHexadecimalToBigint(hex)
+export function hexadecimalToBigInt(hex: string): bigint {
+    return beHexadecimalToBigInt(hex)
 }
 
 /**
- * Converts a buffer to a bigint by interpreting the buffer as a hexadecimal string.
+ * Converts a buffer of bytes to a bigint using big-endian byte order.
  * @param b The buffer to convert.
  * @returns The bigint representation of the buffer's contents.
  */
-export function beBufferToBigint(b: Buffer): bigint {
+export function beBufferToBigInt(b: Buffer): bigint {
     return BigInt(`0x${b.toString("hex")}`)
 }
 
 /**
- * Converts a buffer to a bigint. Alias for beBufferToBigint.
- * @param b - The buffer to convert.
+ * Converts a buffer to a bigint using little-endian byte order.
+ * @param buffer The buffer to convert.
+ * @returns The bigint representation of the buffer's contents in little-endian.
+ */
+export function leBufferToBigInt(buffer: Buffer): bigint {
+    return BigInt(`0x${Buffer.from(buffer).reverse().toString("hex")}`)
+}
+
+/**
+ * Converts a buffer to a bigint. Alias for beBufferToBigInt.
+ * @param b The buffer to convert.
  * @returns The bigint representation of the buffer's contents.
  */
-export function bufferToBigint(b: Buffer): bigint {
-    return beBufferToBigint(b)
+export function bufferToBigInt(b: Buffer): bigint {
+    return beBufferToBigInt(b)
 }
 
 /**
- * Converts a bigint to a buffer, assuming big-endian byte order, and fills with zeros if necessary.
+ * Converts a bigint to a buffer and fills with zeros if necessary.
+ * It uses big-endian byte order.
  * @param n The bigint to convert.
- * @param length The number of bytes of the buffer to return.
+ * @param size The number of bytes of the buffer to return.
```

Comment:
You should document what happens if this parameter is absent.  Also document the error thrown if the size is too small.
I like the new version of the argument and logic, thanks.